myapp/app/helpers/maintenance_helper.rb

```diff
@@ -0,0 +1,8 @@
+module MaintenanceHelper
+
+  def restart_server
+    pid_file = Rails.root.join('tmp/pids/server.pid')
+    system "kill -s USR1 $(cat #{pid_file})"
```

Comment:
https://github.com/puma/puma/blob/master/docs/signals.md#puma-signals

> USR1 restart workers in phases, a rolling restart. This will not reload configuration file.

と書いてありますが、USR1を送ったら、`config/application.rb`は実行されて、`tmp/maintenance.yml`は配置されました。
*configuration file*とは別物ですかね？ 🤔 

とりあえずrolling restartにします。

Reply:
検証してないので全く自信ありませんが、ドキュメント通りUSR1シグナル送って `config/application.rb` が再適用されたのであれば、ここでいう configuration file は `config/puma.rb` ということなのかなと思いました。（`config/*` ということではない）